python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -543,6 +554,7 @@ def run(
         raise_on_error: bool,
         context: Optional[OpExecutionContext],
         adapter: Optional[BaseAdapter],
+        fetch_table_metadata: bool,
```

Comment:
A little worried about booleans like this getting out of control. Not entirely sure if this is better, but would something like the following be feasible:

```python
@dbt_assets
def my_dbt_assets(dbt):
    yield from dbt.run(...).fetch_row_counts()
```

More verbose but also more composable. And would potentially offer the ability to add more control over what subset of models we fetch row counts for, if fetching them all is too expensive.

Reply:
I agree. I don't particularly like the boolean pattern myself.

Since `DbtCliInvocation` is the intermediate object, it makes sense to let it be the object that dictates how the default translation from dbt events to Dagster events happens.

Just making this a bit more explicit: `.stream()` should still be called, but only when `DbtCliInvocation` is finalized. `.fetch_XXX` are methods on `DbtCliInvocation` that return `Self`.

```python
@dbt_assets(...)
def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
    yield from (
        dbt.run(...)
        .fetch_column_schema() # Fetch column schema in Python, rather than in dbt jinja 
        .fetch_row_counts()    # This PR
        .stream()
    )